packages/components/src/playground/button.stories.tsx

```diff
@@ -1,30 +1,35 @@
 import * as React from "react";
-
 import Button from "./button";
-import { action } from "@storybook/addon-actions";
+import { Story } from "@storybook/react/types-6-0";
+
+type Args = React.ComponentProps<typeof Button>;
 
 export default {
   title: "ButtonPlayground",
   component: Button,
 };
 
-export const text = (): JSX.Element => (
-  <Button onClick={action("clicked")}>Hello Button</Button>
-);
+const Template: Story<Args> = (args) => <Button {...args} />;
+
+export const text = Template.bind(null);
+text.args = {
+  children: "Hello Button",
+};
```

Comment:
A downside I've noticed to component story format is we lose type safety. There's nothing here enforcing that we've actually provided all necessary args.

Haven't come up with a good solution for that, yet.

Reply:
can we do `text.args: ButtonProps = ...`?

Reply:
Unfortunately, that doesn't work

Reply:
We may have to pass the args through a function or something to get type safety

```ts
text.args = assignArgs<Args>({
  children: "Hello",
});
```

Or maybe pass the whole thing through a function

```ts
export const text = makeStory(template, {
  children: "Hello Button",
});
```